snapshot.go

```diff
@@ -311,6 +303,8 @@ func (d *DB) makeEventuallyFileOnlySnapshot(
 // call.
 //
 // d.mu must be held when calling this method.
+//
+// TODO(bananabrick): Must we also hold the log lock?
```

Comment:
This is unnecessary because we only need the db lock to be able to close the snapshot at the same time as us doing the transition, which doesn't require us to finish installing a version.